desktop-app/app/reducers/browser.js

```diff
@@ -517,6 +521,28 @@ export default function browser(
           proxy: action.profile,
         },
       };
+    case TOGGLE_ALL_DEVICES_DESIGN_MODE:
+      const nextDevices = state.devices;
```

Comment:
I believe we shouldn't be modifying state here. Also, since it's toggle, can we rely on the previous state instead of an action value? Can we do like,

```
const nextDevices = [...state.devices];
nextDevices.forEach(d => (d.designMode = !state.allDevicesInDesignMode));
 return {
        ...state,
        allDevicesInDesignMode: !state.allDevicesInDesignMode,
        devices: nextDevices
};
```

Reply:
>rely on the previous state instead of an action value

In order to keep in sync with other comments, I think it is ok to change it

>I believe we shouldn't be modifying state here

doing `const y = [...x]` only copies the array, the items inside `y` are the same (same reference) as in `x`. to do it right you need to deep clone the elements too, and I don't think it is worth it

Reply:
Understood.